File: frontend/src/lib/enums.ts
/** Enum value lists + display metadata, mirrored from the API contract. */
import type {
  DeliveryStatus,
  DeliveryItemStatus,
  IssueStatus,
  UserRole,
  OrgType,
} from '@/types/models'

export const USER_ROLES: UserRole[] = [
  'Administrator',
  'DistributionManager',
  'HubOperator',
  'Vendor',
]

export const ORG_TYPES: OrgType[] = ['Press', 'Hub', 'DistributionUnit', 'Vendor']

export const DELIVERY_STATUSES: DeliveryStatus[] = [
  'Created',
  'Packed',
  'Dispatched',
  'OutForDelivery',
  'Delivered',
  'Terminated',
]

export const ISSUE_STATUSES: IssueStatus[] = [
  'Open',
  'Assigned',
  'InProgress',
  'Resolved',
  'Escalated',
]

type BadgeVariant = 'default' | 'secondary' | 'destructive' | 'outline'

export const DELIVERY_STATUS_VARIANT: Record<DeliveryStatus, BadgeVariant> = {
  Created: 'secondary',
  Packed: 'secondary',
  Dispatched: 'default',
  OutForDelivery: 'default',
  Delivered: 'outline',
  Terminated: 'destructive',
}

export const DELIVERY_STATUS_LABEL: Record<DeliveryStatus, string> = {
  Created: 'Created',
  Packed: 'Packed',
  Dispatched: 'Dispatched',
  OutForDelivery: 'Out for delivery',
  Delivered: 'Delivered',
  Terminated: 'Terminated',
}

export const ITEM_STATUS_VARIANT: Record<DeliveryItemStatus, BadgeVariant> = {
  Pending: 'secondary',
  Confirmed: 'outline',
  Discrepancy: 'destructive',
  Missed: 'destructive',
}

export const ISSUE_STATUS_VARIANT: Record<IssueStatus, BadgeVariant> = {
  Open: 'destructive',
  Assigned: 'default',
  InProgress: 'default',
  Resolved: 'outline',
  Escalated: 'destructive',
}

File: backend/app/models/enums.py
import enum


class UserRole(str, enum.Enum):
    Administrator = "Administrator"
    DistributionManager = "DistributionManager"
    HubOperator = "HubOperator"
    Vendor = "Vendor"


class OrgType(str, enum.Enum):
    Press = "Press"
    Hub = "Hub"
    DistributionUnit = "DistributionUnit"
    Vendor = "Vendor"


class EntityType(str, enum.Enum):
    USER = "USER"
    ORG = "ORG"


class DeliveryType(str, enum.Enum):
    Delivery = "Delivery"
    Handend = "Handend"


class DeliveryStatus(str, enum.Enum):
    Created = "Created"
    Packed = "Packed"
    Dispatched = "Dispatched"
    OutForDelivery = "OutForDelivery"
    Delivered = "Delivered"
    Terminated = "Terminated"


# Legal forward transitions for the canonical delivery lifecycle.
# Terminated is reachable from any non-terminal state.
DELIVERY_FORWARD_ORDER = [
    DeliveryStatus.Created,
    DeliveryStatus.Packed,
    DeliveryStatus.Dispatched,
    DeliveryStatus.OutForDelivery,
    DeliveryStatus.Delivered,
]

DELIVERY_TERMINAL_STATES = {DeliveryStatus.Delivered, DeliveryStatus.Terminated}


class DeliveryItemStatus(str, enum.Enum):
    Pending = "Pending"
    Confirmed = "Confirmed"
    Discrepancy = "Discrepancy"
    Missed = "Missed"


class IssueStatus(str, enum.Enum):
    Open = "Open"
    Assigned = "Assigned"
    InProgress = "InProgress"
    Resolved = "Resolved"
    Escalated = "Escalated"


class AssignmentAction(str, enum.Enum):
    DRIVER_ASSIGNED = "DRIVER_ASSIGNED"
    DRIVER_CHANGED = "DRIVER_CHANGED"
    DRIVER_REMOVED = "DRIVER_REMOVED"
    VEHICLE_ASSIGNED = "VEHICLE_ASSIGNED"
    VEHICLE_CHANGED = "VEHICLE_CHANGED"
    VEHICLE_REMOVED = "VEHICLE_REMOVED"


class IssueLogAction(str, enum.Enum):
    ISSUE_CREATED = "ISSUE_CREATED"
    ASSIGNED = "ASSIGNED"
    REASSIGNED = "REASSIGNED"
    COMMENT_ADDED = "COMMENT_ADDED"
    STATUS_CHANGED = "STATUS_CHANGED"
    ESCALATED = "ESCALATED"
    RESOLVED = "RESOLVED"


class AccessLevel(str, enum.Enum):
    READ = "READ"
    WRITE = "WRITE"
    CONFIRM = "CONFIRM"


class NotificationSeverity(str, enum.Enum):
    INFO = "INFO"
    WARNING = "WARNING"
    CRITICAL = "CRITICAL"
